cmd/collector/app/handler.go

```diff
@@ -56,16 +63,16 @@ func (aH *APIHandler) RegisterRoutes(router *mux.Router) {
 }
 
 func (aH *APIHandler) saveSpan(w http.ResponseWriter, r *http.Request) {
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	r.Body.Close()
+	if err != nil {
+		http.Error(w, fmt.Sprintf(unableToReadBodyErrFormat, err), http.StatusBadRequest)
```

Comment:
Bad request, because the readAll fails when it it's unable to allocate memory for a buffer. 
Also, this is how `jaeger.thrift` handled reading the body.